src/AuctionHouseBotConfig.h

```diff
@@ -0,0 +1,169 @@
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Define.h"
+#include "Duration.h"
+#include <array>
+
+enum AHItemQualities
+{
+    AHB_ITEM_QUALITY_POOR = 7,      // GREY
+    AHB_ITEM_QUALITY_NORMAL,        // WHITE
+    AHB_ITEM_QUALITY_UNCOMMON,      // GREEN
+    AHB_ITEM_QUALITY_RARE,          // BLUE
+    AHB_ITEM_QUALITY_EPIC,          // PURPLE
+    AHB_ITEM_QUALITY_LEGENDARY,     // ORANGE
+    AHB_ITEM_QUALITY_ARTIFACT,      // LIGHT YELLOW
+};
+
+constexpr uint32 AHB_MAX_DEFAULT_QUALITY = 6;
+constexpr uint32 AHB_DEFAULT_QUALITY_SIZE = AHB_MAX_DEFAULT_QUALITY + 1;
+constexpr uint32 AHB_MAX_QUALITY = AHB_ITEM_QUALITY_ARTIFACT + 1;
+
+class AHBConfig
+{
+public:
+    AHBConfig() = default;
+    ~AHBConfig() = default;
+
+    AHBConfig(uint32 ahid);
+
+    inline uint32 GetAuctionHouseID()
+    {
+        return _auctionHouseID;
+    }
+
+    inline uint32 GetAuctionHouseFactionID()
+    {
+        return _auctionHouseFactionID;
+    }
+
+    inline void SetMinItems(uint32 value)
+    {
+        _minItems = value;
+    }
+
+    uint32 GetMinItems();
+
+    inline void SetMaxItems(uint32 value)
+    {
+        _maxItems = value;
+        // CalculatePercents() needs to be called, but only if
+        // SetPercentages() has been called at least once already.
+    }
+
+    inline uint32 GetMaxItems()
+    {
+        return _maxItems;
+    }
+
+    void SetPercentages(std::array<uint32, AHB_MAX_QUALITY>& percentages);
+
+    uint32 GetPercentages(uint32 color);
+
+    void SetMinPrice(uint32 color, uint32 value);
+
+    uint32 GetMinPrice(uint32 color);
+
+    void SetMaxPrice(uint32 color, uint32 value);
+
+    uint32 GetMaxPrice(uint32 color);
+
+    void SetMinBidPrice(uint32 color, uint32 value);
+
+    uint32 GetMinBidPrice(uint32 color);
+
+    void SetMaxBidPrice(uint32 color, uint32 value);
+
+    uint32 GetMaxBidPrice(uint32 color);
+
+    void SetMaxStack(uint32 color, uint32 value);
+
+    uint32 GetMaxStack(uint32 color);
+
+    void SetBuyerPrice(uint32 color, uint32 value);
+
+    uint32 GetBuyerPrice(uint32 color);
+
+    inline void SetBiddingInterval(Minutes value)
+    {
+        _buyerBiddingInterval = value;
+    }
+
+    inline Minutes GetBiddingInterval()
+    {
+        return _buyerBiddingInterval;
+    }
+
+    void CalculatePercents();
```

Comment:
That is a bad name
This doesn't calculate "Percents", it calculates the maximum number of items per quality level.
I will probably rename to "CalculateMaxCounts" later

Reply:
In general naming is bad all over the place.
"SetMinPrice" doesn't set the min price. It sets the minimum price multiplier in percent with 0-100 value.